In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from wrangle import get_db_url

In [2]:
sql_query = """
        SELECT *
        from properties_2017
        join propertylandusetype using (propertylandusetypeid)
        
        where	propertylandusetypeid = '261';
                """

In [28]:
df = pd.read_sql(sql_query, get_db_url("zillow"))
#df = df.dropna()

In [30]:
pd.set_option('display.max_columns', None)
df.head()

propertylandusetypeid  id  parcelid  airconditioningtypeid  \
0                  261.0   1  10759547                    NaN   
1                  261.0  15  11193347                    NaN   
2                  261.0  16  11215747                    NaN   
3                  261.0  17  11229347                    NaN   
4                  261.0  20  11324547                    1.0   

   architecturalstyletypeid  basementsqft  bathroomcnt  bedroomcnt  \
0                       NaN           NaN          0.0         0.0   
1                       NaN           NaN          0.0         0.0   
2                       NaN           NaN          0.0         0.0   
3                       NaN           NaN          0.0         0.0   
4                       NaN           NaN          2.0         4.0   

   buildingclasstypeid  buildingqualitytypeid  calculatedbathnbr  decktypeid  \
0                  NaN                    NaN                NaN         NaN   
1                  NaN                    NaN                NaN         NaN   
2                  NaN                    NaN                NaN         NaN   
3                  NaN                    NaN                NaN         NaN   
4                  NaN                    8.0                2.0         NaN   

   finishedfloor1squarefeet  calculatedfinishedsquarefeet  \
0                       NaN                           NaN   
1                       NaN                           NaN   
2                       NaN                           NaN   
3                       NaN                           NaN   
4                       NaN                        3633.0   

   finishedsquarefeet12  finishedsquarefeet13  finishedsquarefeet15  \
0                   NaN                   NaN                   NaN   
1                   NaN                   NaN                   NaN   
2                   NaN                   NaN                   NaN   
3                   NaN                   NaN                   NaN   
4                3633.0                   NaN                   NaN   

   finishedsquarefeet50  finishedsquarefeet6    fips  fireplacecnt  \
0                   NaN                  NaN  6037.0           NaN   
1                   NaN                  NaN  6037.0           NaN   
2                   NaN                  NaN  6037.0           NaN   
3                   NaN                  NaN  6037.0           NaN   
4                   NaN                  NaN  6037.0           NaN   

   fullbathcnt  garagecarcnt  garagetotalsqft  hashottuborspa  \
0          NaN           NaN              NaN             NaN   
1          NaN           NaN              NaN             NaN   
2          NaN           NaN              NaN             NaN   
3          NaN           NaN              NaN             NaN   
4          2.0           NaN              NaN             NaN   

   heatingorsystemtypeid    latitude    longitude  lotsizesquarefeet  poolcnt  \
0                    NaN  34140430.0 -118625364.0             4083.0      NaN   
1                    NaN  34585014.0 -118162010.0            11975.0      NaN   
2                    NaN  34563376.0 -118019104.0             9403.0      NaN   
3                    NaN  34526913.0 -118050581.0             3817.0      NaN   
4                    2.0  34560018.0 -118169806.0             9826.0      NaN   

   poolsizesum  pooltypeid10  pooltypeid2  pooltypeid7  \
0          NaN           NaN          NaN          NaN   
1          NaN           NaN          NaN          NaN   
2          NaN           NaN          NaN          NaN   
3          NaN           NaN          NaN          NaN   
4          NaN           NaN          NaN          NaN   

  propertycountylandusecode propertyzoningdesc  rawcensustractandblock  \
0                      0109             LCA11*            6.037800e+07   
1                      0100              PDA1*            6.037910e+07   
2                      0100             PDA21*

In [29]:
df.info(show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2152863 entries, 0 to 2152862
Data columns (total 60 columns):
 #   Column                        Non-Null Count    Dtype  
---  ------                        --------------    -----  
 0   propertylandusetypeid         2152863 non-null  float64
 1   id                            2152863 non-null  int64  
 2   parcelid                      2152863 non-null  int64  
 3   airconditioningtypeid         513889 non-null   float64
 4   architecturalstyletypeid      2393 non-null     float64
 5   basementsqft                  1538 non-null     float64
 6   bathroomcnt                   2152852 non-null  float64
 7   bedroomcnt                    2152852 non-null  float64
 8   buildingclasstypeid           1 non-null        float64
 9   buildingqualitytypeid         1420228 non-null  float64
 10  calculatedbathnbr             2139190 non-null  float64
 11  decktypeid                    12571 non-null    float64
 12  finishedfloor1squarefeet    

In [32]:
df.shape

(2152863, 60)

**TAKEAWAYS**
- Raw dataset has 2152863 rows and  60 columns
- Need to `poolcnt` fill null values with 0 
- Need to fill `garagecarcnt` null values with 0
- Need to fill `fireplacecnt` null values with 0
- Need to fill `hashottuborspa` null values with 0

- Drop rows with null values in `calculatedfinishedsquarefeet`
- Drop rows with null values in `taxamount`
- Drop rows with null val

In [31]:
df.isna().mean()


propertylandusetypeid           0.000000e+00
id                              0.000000e+00
parcelid                        0.000000e+00
airconditioningtypeid           7.612997e-01
architecturalstyletypeid        9.988885e-01
basementsqft                    9.992856e-01
bathroomcnt                     5.109475e-06
bedroomcnt                      5.109475e-06
buildingclasstypeid             9.999995e-01
buildingqualitytypeid           3.403073e-01
calculatedbathnbr               6.351078e-03
decktypeid                      9.941608e-01
finishedfloor1squarefeet        9.240189e-01
calculatedfinishedsquarefeet    3.940799e-03
finishedsquarefeet12            7.615440e-03
finishedsquarefeet13            9.999981e-01
finishedsquarefeet15            9.999944e-01
finishedsquarefeet50            9.240189e-01
finishedsquarefeet6             9.963314e-01
fips                            0.000000e+00
fireplacecnt                    8.664123e-01
fullbathcnt                     6.351078e-03
garagecarc

In [25]:
df = df.drop(columns=['id', 'parcelid', 'basementsqft', 'buildingclasstypeid', 'calculatedbathnbr', 'decktypeid',  ])

KeyError: "['id', 'parcelid'] not found in axis"

In [23]:
for column in df.columns:
    print(column)
    print(df[column].value_counts())
    print("-----------------")

propertylandusetypeid
261.0    2152863
Name: propertylandusetypeid, dtype: int64
-----------------
airconditioningtypeid
1.0     447937
13.0     56602
5.0       8387
11.0       901
12.0        46
9.0         11
3.0          5
Name: airconditioningtypeid, dtype: int64
-----------------
architecturalstyletypeid
7.0     2124
2.0       95
21.0      76
8.0       45
3.0       42
5.0       10
10.0       1
Name: architecturalstyletypeid, dtype: int64
-----------------
basementsqft
1528.0    25
240.0     21
100.0     17
144.0     16
360.0     15
          ..
416.0      1
618.0      1
575.0      1
568.0      1
136.0      1
Name: basementsqft, Length: 748, dtype: int64
-----------------
bathroomcnt
2.00     943589
3.00     422841
1.00     414324
2.50     142981
4.00      82155
1.50      31211
3.50      28518
5.00      28362
4.50      19506
0.00      13027
6.00      10747
5.50       6217
7.00       4394
8.00       1692
6.50       1333
9.00        713
7.50        384
10.00       325
11.00       146